client/src/FredChart.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react';
+import { Chart, Chart as PrimeChart } from 'primereact/chart';
+import { Dropdown } from 'primereact/dropdown';
+import axios from 'axios';
+
+
+function FredChart() {
+    const [chartData, setChartData] = useState(null);
+    const [selectedSeries, setSelectedSeries] = useState('CPIAUCSL'); // default series
+    const [loading, setLoading] = useState(false);
+
+    const seriesOptions = [
+        { label: 'Consumer Price Index (CPI)', value: 'CPIAUCSL' },
+        { label: 'Unemployment Rate', value: 'UNRATE' },
+        { label: 'Real GDP', value: 'GDPC1' },
+        { label: 'Federal Funds Rate', value: 'FEDFUNDS' }
+    ];
+
+    const fetchData = async (seriesId) => {
+        setLoading(true);
+        try {
+            const res = await axios.get(`http://localhost:8000/api/v1/fred/observations/${seriesId}?sort_order=asc`);
```

Comment:
```suggestion
            const res = await axios.get(`API_BASE_URLapi/v1/fred/observations/${seriesId}?sort_order=asc`);
```
Instead of hardcoding localhost:8000, we should use an environment variable with a fallback to localhost:8000 if none is defined.
`const API_BASE_URL = process.env.REACT_APP_API_BASE_URL || 'http://localhost:8000';
`